wallet/wallet.go

```diff
@@ -3414,6 +3414,14 @@ func (w *Wallet) publishTransaction(tx *wire.MsgTx) (*chainhash.Hash, error) {
 	}
 
 	txid, err := chainClient.SendRawTransaction(tx, false)
+
+	// Determine if this was an RPC error thrown due to the transaction
+	// already confirming.
+	var rpcTxConfirmed bool
+	if rpcErr, ok := err.(*btcjson.RPCError); ok {
```

Comment:
Possible to add a unit test for this behavior?

Reply:
It's somewhat already covered [here](https://github.com/lightningnetwork/lnd/blob/master/lnwallet/interface_test.go#L1435).

Reply:
I was thinking a `btcwallet` unit test with a mocked `chainClient`. We could make the mock mimic the behavior of various backends/versions, and ensure we handle the diff errors correctly.

Reply:
Since we're doing string matching I'm not sure a unit test with a mock would work here. We'd just be including the same strings in there, which would seem to me like we're testing `string.Contains`.